reana/reana_dev/git.py

```diff
@@ -371,8 +372,30 @@ def git_clone(user, component, exclude_components):  # noqa: D301
     if exclude_components:
         exclude_components = exclude_components.split(",")
     components = select_components(component, exclude_components)
+
+    if "--target-directory" in sys.argv:
```

Comment:
It's not needed to check against `sys.argv`. `click` already offers the capability to check its value directly, and since the default value is `.` there will always be a `target_directory`.

I believe this check can be better done in a [callback validator](https://click.palletsprojects.com/en/8.0.x/options/#callbacks-for-validation).

Reply:
I agree, a callback validator is a better option